checkov/yaml_doc/base_registry.py

```diff
@@ -44,10 +44,13 @@ def _scan_yaml_array(
                         )
             if isinstance(analyzed_entities, list):
                 for item in analyzed_entities:
+                    item_dict = item
+                    if isinstance(item, str):
+                        item_dict = self.set_lines_for_item(item)
```

Comment:
did you do this, because of typing issues? instead of just
```suggestion
                    if isinstance(item, str):
                        item = self.set_lines_for_item(item)
```